### PR TITLE
feat(x/ecocredit): codec and server registration

### PR DESCRIPTION
**Pull request by aaronc**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by technicallyty — APPROVED** [review]

File: x/ecocredit/basket/codec.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+)
+
+func RegisterTypes(registry codectypes.InterfaceRegistry) {
+	msgservice.RegisterMsgServiceDesc(registry, &_Msg_serviceDesc)
+}
```
Comment (robert-zaremba) [inline x/ecocredit/basket/codec.go:11]: we should add here the message interfaces as well.

File: x/ecocredit/basket/codec.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&MsgCreate{}, "regen.basket/MsgCreate", nil)
+	cdc.RegisterConcrete(&MsgPut{}, "regen.basket/MsgPut", nil)
+	cdc.RegisterConcrete(&MsgTake{}, "regen.basket/MsgTake", nil)
```
Comment (robert-zaremba) [inline x/ecocredit/basket/codec.go:16]: looks like we need to update `Type` function

File: x/ecocredit/basket/codec.go (hunk 1 above)
Comment (aaronc) [reply x/ecocredit/basket/codec.go:11]: Pretty sure that's taken care of automatically by this function

File: x/ecocredit/basket/codec.go (hunk 2 above)
Comment (aaronc) [reply x/ecocredit/basket/codec.go:16]: Which Type function?

Comment (aaronc) [reply x/ecocredit/basket/codec.go:16]: You mean Msg.Type? I'm pretty sure that's no longer used by the SDK but sure

Comment (aaronc) [reply x/ecocredit/basket/codec.go:16]: Yeah `Msg.Type` and `Msg.Route` can just be ignored. The SDK doesn't use them. We can just as well panic

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**